rclpy/src/rclpy/_rclpy.c

```diff
@@ -1588,6 +1588,48 @@ rclpy_wait_set_add_entity(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Destroy the waitset structure
+/**
+ * \param[in] pywait_set Capsule pointing to the waitset structure
+ * \return NULL
+ */
+static PyObject *
+rclpy_destroy_wait_set(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pywait_set;
+
+  if (!PyArg_ParseTuple(args, "O", &pywait_set)) {
+    return NULL;
+  }
+  void * p = PyCapsule_GetPointer(pywait_set, NULL);
+  if (p == Py_None) {
+    PyErr_Format(PyExc_RuntimeError, "wait set is None");
+    return NULL;
+  }
+  rcl_wait_set_t * wait_set = (rcl_wait_set_t *)p;
+  if (!wait_set) {
+    PyErr_Format(PyExc_RuntimeError, "wait set is NULL");
+    return NULL;
+  }
+
+  rcl_ret_t ret = rcl_wait_set_fini(wait_set);
+  if (ret != RCL_RET_OK) {
+    PyErr_Format(PyExc_RuntimeError,
+      "Failed to fini wait set: %s", rcl_get_error_string_safe());
+    rcl_reset_error();
+    return NULL;
+  }
+
+  PyMem_Free(wait_set);
+
+  if (PyCapsule_SetPointer(pywait_set, Py_None)) {
+    // exception set by PyCapsule_SetPointer
+    return NULL;
+  }
+
+  Py_RETURN_NONE;
```

Comment:
This could return true on success and false on failure instead of always returning none (thouhg it looks like every failure raises an exception so I'm not sure we would ever return false)

Reply:
Since the function always raises in case of problems it doesn't need to return anything. This is also the behavior of many other function in this C extension.

Reply:
:+1: